ports/nrf/boards/arduino_nano_33_ble/modules/lsm9ds1.py

```diff
@@ -0,0 +1,187 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2013, 2014 Damien P. George
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+
+
+LSM9DS1 - 9DOF inertial sensor of STMicro driver for MicroPython.
+The sensor contains an accelerometer / gyroscope / magnetometer
+Uses the internal FIFO to store up to 16 gyro/accel data, use the iter_accel_gyro generator to access it.
+
+Example usage:
+import time
+from lsm9ds1 import LSM9DS1
+from machine import Pin, I2C
+
+lsm = LSM9DS1(I2C(1, scl=Pin(15), sda=Pin(14)))
+
+while (True):
+    #for g,a in lsm.iter_accel_gyro(): print(g,a)    # using fifo
+    print('Accelerometer: x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.accel()))
+    print('Magnetometer:  x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.magnet()))
+    print('Gyroscope:     x:{:>8.3f} y:{:>8.3f} z:{:>8.3f}'.format(*lsm.gyro()))
+    print("")
+    time.sleep_ms(100)
+"""
+import array
+
+
+class LSM9DS1:
+    WHO_AM_I = const(0xF)
+    CTRL_REG1_G = const(0x10)
+    INT_GEN_SRC_G = const(0x14)
+    OUT_TEMP = const(0x15)
+    OUT_G = const(0x18)
+    CTRL_REG4_G = const(0x1E)
+    STATUS_REG = const(0x27)
+    OUT_XL = const(0x28)
+    FIFO_CTRL_REG = const(0x2E)
+    FIFO_SRC = const(0x2F)
+    OFFSET_REG_X_M = const(0x05)
+    CTRL_REG1_M = const(0x20)
+    OUT_M = const(0x28)
+    SCALE_GYRO = [(245, 0), (500, 1), (2000, 3)]
+    SCALE_ACCEL = [(2, 0), (4, 2), (8, 3), (16, 1)]
```

Comment:
These constants are better in the global scope, prefixed with `_`.

Also, `SCALE_GYRO` and `SCALE_ACCEL` can be constants as well if they are tuples of tuples of ints.